src/dowel/csv_output.py

```diff
@@ -48,7 +51,25 @@ def record(self, data, prefix=''):
                            'Did you change key sets after your first '
                            'logger.log(TabularInput)?'.format(
                                set(self._fieldnames), set(to_csv.keys())))
-
+                with open(self._filename, "r") as old:
+                    old_data = csv.DictReader(old)
+                    
+                    # Update the header
+                    self._fieldnames = set(self._longestHeader)|set(to_csv.keys())
+                    self._longestHeader = self._fieldnames
+
+                    # Change the writer with the new keys
+                    self._writer = csv.DictWriter(
+                        self._log_file,
+                        fieldnames=self._fieldnames,
+                        extrasaction='ignore')
```

Comment:
Why use `extrasaction='ignore'` here? `extrasaction='raise'` (the default), seems like a safer option.

Reply:
Since the new filenames will always contain all keys, I put the 'ignore' here. However, 'raise' will always be a much safer option. Thank you for pointing out!